pkg/controller/ipam/validate.go

```diff
@@ -107,6 +145,60 @@ func humanReadableIPRanges(ipRanges []crdv1alpha2.SubnetIPRange) string {
 	return fmt.Sprintf("[%s]", strings.Join(strs, ","))
 }
 
+func rangesOverlap(r1, r2 crdv1alpha2.SubnetIPRange) bool {
+	if r1.CIDR == "" {
+		if r2.CIDR == "" {
+			r1start := net.ParseIP(r1.Start).To16()
+			r1end := net.ParseIP(r1.End).To16()
+			r2start := net.ParseIP(r2.Start).To16()
+			r2end := net.ParseIP(r2.End).To16()
+
+			if bytes.Compare(r2start, r1start) >= 0 && bytes.Compare(r2start, r1end) <= 0 {
+				return true
+			} else if bytes.Compare(r1start, r2start) >= 0 && bytes.Compare(r1start, r2end) <= 0 {
+				return true
+			}
+		} else {
+			_, cidr2, _ := net.ParseCIDR(r2.CIDR)
+			return cidr2.Contains(net.ParseIP(r1.Start)) || cidr2.Contains(net.ParseIP(r1.End))
```

Comment:
This cannot detect the case that the CIDR is contained by the start-end range. For example:
r1 = 1.1.1.1-1.1.1.10, r2 = 1.1.1.4/30